### PR TITLE
Update etcd to 2.2.1

### LINKED ISSUE #16618: Update default etcd version
Right now we have 2.0.12 set in quite a few places.  This is from June.  Probably time to update to something newer.  Now is a good time to do it as we can flush out any issues before v1.2 is released.

### PR DESCRIPTION
**Pull request by jbeda**
This fixes #16618.

I've run the e2e tests locally and have success assuming #16676.

### CODE REVIEW

**Comment by ixdy** [comment]
Probably should update `cluster/mesos/docker/test/bin/install-etcd.sh` and `docs/devel/development.md` too.

**Comment by jbeda** [comment]
I've updated `cluster/mesos/docker/test/bin/install-etcd.sh` but don't see anything in `docs/devel/development.md` that needs to be updated.

**Comment by ixdy** [comment]
You're right, I forgot I also did https://github.com/kubernetes/kubernetes/commit/1c5f252653a5a5559b1f58ea1eb9e0ddfcc00c02 which removed the version-specific text.

File: cluster/images/etcd/.gitignore [hunk 1]
```diff
@@ -0,0 +1,4 @@
+etcd
+etcdctl
+etcd-*-linux-amd64.tar.gz
+etcd-*-linux-amd64/
```
Comment (wojtek-t) [inline cluster/images/etcd/.gitignore]: nit - can you please remove this funny character from the end of line?

**Comment by wojtek-t** [comment]
LGTM - just one minor comment

File: cluster/images/etcd/.gitignore (hunk 1 above)
Comment (jbeda) [inline cluster/images/etcd/.gitignore]: The trailing slash?  That specifies that it only matches a directory but not a file. https://git-scm.com/docs/gitignore

Comment (ixdy) [inline cluster/images/etcd/.gitignore]: I think @wojtek-t was referring to the weird circle with slash that appears in GitHub's diff view. Hovering over indicates this means there's no newline at the end of this file, so maybe add one?

Comment (wojtek-t) [inline cluster/images/etcd/.gitignore]: Yes - I meant what @ixdy wrote

**Comment by jbeda** [comment]
PTAL - looks like the @k8s-bot failure was due to not being able to reach docker on the build machine.

**Comment by wojtek-t** [comment]
LGTM - thanks!

**Comment by ixdy** [comment]
software is the worst.

@k8s-bot test this
@k8s-bot unit test this

**Comment by ixdy** [comment]
@k8s-bot test this please

**Comment by piosz** [comment]
@k8s-bot e2e test this

**Pull request merged**